instrumentation/aws-sdk/aws-sdk-1.11/library/src/main/java/io/opentelemetry/instrumentation/awssdk/v1_11/AwsSdkAttributesExtractor.java

```diff
@@ -64,6 +68,7 @@ public void onEnd(
       @Nullable Throwable error) {
     if (response != null) {
       Object awsResp = response.getAwsResponse();
+      setAttribute(attributes, AWS_SECRETSMANAGER_SECRET_ARN, awsResp, RequestAccess::getSecretArn);
```

Comment:
Perhaps there should be a null check for `awsResp` so you wouldn't need to do it in `RequestAccess`.
One thing that is confusing for me is that in case of secret manager you test `CreateSecretRequest` so it is understandable that the arn is on the response. For step functions you test `DescribeStateMachineRequest` and `DescribeActivityRequest` there in a real application shouldn't the arn already be set on the request when running these agains real aws service? I guess there is also `CreateStateMachineRequest` where you'd need to read the arn from the response.